pandas/core/generic.py

```diff
@@ -917,7 +924,24 @@ def swaplevel(self: FrameOrSeries, i=-2, j=-1, axis=0) -> FrameOrSeries:
     # ----------------------------------------------------------------------
     # Rename
 
-    def rename(self, *args, **kwargs):
+    def rename(
+        self,
+        mapper: Optional[
+            Union[Mapping[Hashable, Any], Callable[[Hashable], Hashable]]
+        ] = None,
+        *,
+        index: Optional[
+            Union[Mapping[Hashable, Any], Callable[[Hashable], Hashable]]
+        ] = None,
+        columns: Optional[
+            Union[Mapping[Hashable, Any], Callable[[Hashable], Hashable]]
+        ] = None,
+        axis: Optional[Axis] = None,
+        copy: bool = True,
+        inplace: bool = False,
+        level: Optional[Level] = None,
+        errors: str = "ignore",
+    ):
```

Comment:
return type.

Reply:
Couldn't get this to work for some reason. When I had `Optional[FrameOrSeries]` I would get `pandas/core/generic.py:1107: error: Incompatible return value type (got "NDFrame", expected "Optional[FrameOrSeries]")`

I then changed to `Optional[NDFrame]` but got `pandas/core/frame.py:4110: error: Incompatible return value type (got "Optional[NDFrame]", expected "Optional[DataFrame]")`

Any ideas?

Reply:
you also need to annotate self as FrameOrSeries to bind the typevar

Reply:
Ah...didn't realize that would matter. Nice catch!